bundles/org.openhab.binding.gce/src/main/java/org/openhab/binding/gce/internal/handler/Ipx800DeviceConnector.java

```diff
@@ -0,0 +1,171 @@
+    private static final int DEFAULT_SOCKET_TIMEOUT = 5000;
+    private static final int DEFAULT_RECONNECT_TIMEOUT = 5000;
+    private static final int MAX_KEEPALIVE_FAILURE = 3;
+    private final static String ENDL = "\r\n";
+
+    private final String hostname;
+    public final int portNumber;
+    private Optional<Ipx800MessageParser> parser = Optional.empty();
+
+    private boolean interrupted = false;
+    private boolean connected = false;
+
+    private @NonNullByDefault({}) Socket client;
+    private @NonNullByDefault({}) BufferedReader in;
+    private @NonNullByDefault({}) PrintWriter out;
+
+    private int failedKeepalive = 0;
+    private boolean waitingKeepaliveResponse = false;
+
+    public Ipx800DeviceConnector(String hostname, int portNumber) {
+        this.hostname = hostname;
+        this.portNumber = portNumber;
+    }
+
+    public synchronized void send(String message) {
+        logger.debug("Sending '{}' to Ipx800", message);
+        out.write(message + ENDL);
+        out.flush();
+    }
+
+    /**
+     * Connect to the ipx800
+     *
+     * @throws IOException
+     */
+    private void connect() throws IOException {
+        disconnect();
+        logger.debug("Connecting {}:{}...", hostname, portNumber);
+        client = new Socket(hostname, portNumber);
+        client.setSoTimeout(DEFAULT_SOCKET_TIMEOUT);
+        client.getInputStream().skip(client.getInputStream().available());
+        in = new BufferedReader(new InputStreamReader(client.getInputStream()));
+        out = new PrintWriter(client.getOutputStream(), true);
+        connected = true;
+        logger.debug("Connected to {}:{}", hostname, portNumber);
+    }
+
+    /**
+     * Disconnect the device
+     */
+    private void disconnect() {
+        if (connected) {
+            logger.debug("Disconnecting");
+            try {
+                client.close();
+            } catch (IOException e) {
+                logger.warn("Unable to disconnect {}", e.getMessage());
+            }
+            connected = false;
+            logger.debug("Disconnected");
+        }
+    }
+
+    /**
+     * Stop the device thread
+     */
+    public void destroyAndExit() {
+        interrupted = true;
+        disconnect();
+    }
+
+    /**
+     * Send an arbitrary keepalive command which cause the IPX to send an update.
+     * If we don't receive the update maxKeepAliveFailure time, the connection is closed and reopened
+     */
+    private void sendKeepalive() {
+        if (waitingKeepaliveResponse) {
+            failedKeepalive++;
+            logger.debug("Sending keepalive, attempt {}", failedKeepalive);
+        } else {
+            failedKeepalive = 0;
+            logger.debug("Sending keepalive");
+        }
+        out.println("GetIn01");
+        out.flush();
+        waitingKeepaliveResponse = true;
+    }
+
+    @Override
+    public void run() {
+        interrupted = false;
+        while (!interrupted) {
```

Comment:
Why not use Thread's flag?
```suggestion
        while (!interrupted()) {
```